### PR TITLE
Improve error check

### LINKED ISSUE #36: Error when pressing backspace after complete
Hi I see the error in the screenshot when I do TabNine::config and then press backspace after finalizing the complete
 
Screenshot from 2019-08-26 20-32-57

I get it every time. The TabNine is installed minutes ago and I did run the ./install.sh

### PR DESCRIPTION
**Pull request by Shougo**
I have improved error check.

Fix #36

### CODE REVIEW

**Comment by Shougo** [comment]
@tbodt Ping

File: rplugin/python3/deoplete/sources/tabnine.py [hunk 1]
```diff
@@ -66,17 +66,21 @@ def get_complete_position(self, context):
         m = re.search(r'\s+$', context['input'])
         if m:
             return -1
+
         self._response = self._get_response(context)
-        if self._response is None:
+        if not self._response:
```
Comment (tbodt) [inline rplugin/python3/deoplete/sources/tabnine.py]: When would the response be []?

**Review by tbodt — COMMENTED** [review]
Thanks! Sorry I forgot about this.

File: rplugin/python3/deoplete/sources/tabnine.py (hunk 1 above)
Comment (Shougo) [reply rplugin/python3/deoplete/sources/tabnine.py]: OK.  I have changed it.

File: rplugin/python3/deoplete/sources/tabnine.py [hunk 2]
```diff
@@ -66,17 +66,21 @@ def get_complete_position(self, context):
...
             return -1
+
         old_prefix = self._response['old_prefix']
         if not context['input'].endswith(old_prefix):
             return -1
         return len(context['input']) - len(old_prefix)
 
     def gather_candidates(self, context):
-        response = self._response
+        if not self._response:
```
Comment (tbodt) [inline rplugin/python3/deoplete/sources/tabnine.py]: Could it be [] here as well? `is None` is preferred if you want to check for None.

Comment (Shougo) [reply rplugin/python3/deoplete/sources/tabnine.py]: I have changed it.

**Comment by Shougo** [comment]
Thanks

**Pull request merged**